### PR TITLE
Improved Tag Management

### PR DESCRIPTION
**Pull request by Juxarius**
Improved Tag management:
- Tags can now be added 1 by 1 one using the t/ prefix instead of overwriting the entire tag list
- Tags can be deleted using the dt/ prefix followed by the name of the tag to be deleted
- Tags can also be mass deleted using dt/all
- dt/ and t/ prefixed are allowed to be used simultaneously on the same line

Developer sided changes:
- Changed getTags from unmodifiableSet to a normal Set so that i can edit the tags
- Added dt/ prefix to relevant files to accept delete tag commands
- Added tagsToDel in editPersonDescriptor to indicate a change required, include getter and setter methods

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -46,6 +47,9 @@
     public static final String ADDRESS_DESC_BOB = " " + PREFIX_ADDRESS + VALID_ADDRESS_BOB;
     public static final String TAG_DESC_FRIEND = " " + PREFIX_TAG + VALID_TAG_FRIEND;
     public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
+    public static final String DELTAG_DESC_FRIEND = " " + PREFIX_DELTAG + VALID_TAG_FRIEND;
+    public static final String DELTAG_DESC_HUSBAND = " " + PREFIX_DELTAG + VALID_TAG_HUSBAND;
+    public static final String DELTAG_ALL = " " + PREFIX_DELTAG + "all";
```
Comment (OscarWang114) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:52]: Would be better to make `all` a `public static final String`

File: src/test/java/systemtests/DeleteCommandSystemTest.java [hunk 2]
```diff
@@ -108,8 +107,6 @@ public void delete() {
         /* Case: invalid arguments (extra argument) -> rejected */
         assertCommandFailure(DeleteCommand.COMMAND_WORD + " 1 abc", MESSAGE_INVALID_DELETE_COMMAND_FORMAT);
 
-        /* Case: mixed case command word -> rejected */
-        assertCommandFailure("DelETE 1", MESSAGE_UNKNOWN_COMMAND);
```
Comment (OscarWang114) [inline src/test/java/systemtests/DeleteCommandSystemTest.java:112]: Consider replacing with `assertCommandSucces()` case instead of deleting?

**Review by OscarWang114 — COMMENTED** [review]
Great work @Juxarius ! 
I am curious about what will happen if I type`edit 1 t/` in the console?

File: src/test/java/systemtests/DeleteCommandSystemTest.java (hunk 2 above)
Comment (Juxarius) [reply src/test/java/systemtests/DeleteCommandSystemTest.java:112]: Lol omg u are right hahaha

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 1 above)
Comment (Juxarius) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:52]: It seems to be used very specificaly so i thought there wasnt a need to redirect one more time, but ill consider it

**Comment by Juxarius** [comment]
About the t/, i posted an BUG issue. With this feature t/ will not do anything but it does not trigger a missing tag statement even tho the regex suggests +, which means a tag is required

**Review by OscarWang114 — APPROVED** [review]
Approved

**Pull request merged**